test/behavior/bugs/10684.zig

```diff
@@ -0,0 +1,12 @@
+const std = @import("std");
+const expectEqualStrings = std.testing.expectEqualStrings;
+
+test "slicing slices" {
+    const foo = "1234";
```

Comment:
Some of these tests will need to be escaped wrt self-hosted native backends, e.g.:

```zig
if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest;
```

Reply:
Thanks, I will try to skip more in-house backends.